src/dht/provide.js

```diff
@@ -20,6 +21,13 @@ module.exports = (send) => {
       cids = [cids]
     }
 
+    // Validate CID(s) and serialize
+    try {
+      cids = cids.map(cid => new CID(cid).toBaseEncodedString('base58btc'))
```

Comment:
I believe that the DHT will only like CIDs V0 and not CIDs V1. Could you double check with both?

Reply:
I've added a test https://github.com/ipfs/interface-ipfs-core/pull/221/commits/bde22b2db3dd72e895619c5694b5dd6ed14318c7

go-ipfs (0.4.13) does not error when provide a CIDv1:

<img width="417" alt="screen shot 2018-03-07 at 10 11 54" src="https://user-images.githubusercontent.com/152863/37086665-094f6f14-21f0-11e8-8970-4d0ae7f643e8.png">

In js-ipfs, [`repo.blocks.has`](https://github.com/ipfs/js-ipfs/blob/0920ff4712cd979192519deec432d3413fa1a992/src/core/components/dht.js#L119) returns false, so the test fails in https://github.com/ipfs/js-ipfs/pull/1250 but [`libp2pNode.contentRouting.provide`](https://github.com/ipfs/js-ipfs/blob/0920ff4712cd979192519deec432d3413fa1a992/src/core/components/dht.js#L133) does not error.